crates/bdk/src/wallet/mod.rs

```diff
@@ -269,416 +284,625 @@ impl<D> Wallet<D> {
-    /// for secret_key in wallet.get_signers(KeychainKind::External).signers().iter().filter_map(|s| s.descriptor_secret_key()) {
-    ///     // secret_key: tprv8ZgxMBicQKsPe73PBRSmNbTfbcsZnwWhz5eVmhHpi31HW29Z7mc9B4cWGRQzopNUzZUT391DeDJxL2PefNunWyLgqCKRMDkU1s2s8bAfoSk/84'/0'/0'/0/*
-    ///     println!("secret_key: {}", secret_key);
-    /// }
-    ///
-    /// Ok::<(), Box<dyn std::error::Error>>(())
-    /// ```
-    pub fn get_signers(&self, keychain: KeychainKind) -> Arc<SignersContainer> {
-        match keychain {
-            KeychainKind::External => Arc::clone(&self.signers),
-            KeychainKind::Internal => Arc::clone(&self.change_signers),
+        Balance {
+            immature,
+            trusted_pending,
+            untrusted_pending: 0,
+            confirmed,
         }
     }
 
-    /// Start building a transaction.
+    /// Returns the [`Balance`] of a given list of keychains.
     ///
-    /// This returns a blank [`TxBuilder`] from which you can specify the parameters for the transaction.
-    ///
-    /// ## Example
+    /// The `should_trust` predicated can be used if user wishes to categorize the unconfirmed balance among "trusted" and
+    /// "untrusted" sources. This typically useful when a wallet is receiving into *external* keychains and which receives funds
+    /// from external wallets, not controlled by the user.
     ///
-    /// ```
-    /// # use std::str::FromStr;
-    /// # use bitcoin::*;
-    /// # use bdk::*;
-    /// # let descriptor = "wpkh(tpubD6NzVbkrYhZ4Xferm7Pz4VnjdcDPFyjVu5K4iZXQ4pVN8Cks4pHVowTBXBKRhX64pkRyJZJN5xAKj4UDNnLPb5p2sSKXhewoYx5GbTdUFWq/*)";
-    /// # let mut wallet = doctest_wallet!();
-    /// # let to_address = Address::from_str("2N4eQYCbKUHCCTUjBJeHcJp9ok6J2GZsTDt").unwrap();
-    /// let (psbt, details) = {
-    ///    let mut builder =  wallet.build_tx();
-    ///    builder
-    ///        .add_recipient(to_address.script_pubkey(), 50_000);
-    ///    builder.finish()?
-    /// };
-    ///
-    /// // sign and broadcast ...
-    /// # Ok::<(), bdk::Error>(())
-    /// ```
-    ///
-    /// [`TxBuilder`]: crate::TxBuilder
-    pub fn build_tx(&mut self) -> TxBuilder<'_, D, DefaultCoinSelectionAlgorithm, CreateTx> {
-        TxBuilder {
-            wallet: alloc::rc::Rc::new(core::cell::RefCell::new(self)),
-            params: TxParams::default(),
-            coin_selection: DefaultCoinSelectionAlgorithm::default(),
-            phantom: core::marker::PhantomData,
-        }
+    /// When this categorization is not needed, passing in `should_trust : false` will default to all keychain balances being
+    /// "trusted".
+    pub fn get_balance_for_keychains(
+        &self,
+        keychains: impl IntoIterator<Item = K>,
+        should_trust: impl Fn(&K) -> bool,
+    ) -> Balance {
+        keychains.into_iter().fold(Balance::default(), |acc, k| {
+            let mut balance = self.get_balance_by_keychain(&k);
```

Comment:
I'm thinking we can have a boolean parameter on `get_balance_for_keychain`. The API can look like `get_balance_by_keychain(&self, keychain: &K, should_trust: bool)`. This will remove the need for `make_untrusted` and also remove this `if` condition.

Reply:
That's a nice idea.. I also didn't like the awkward `make_untrusted` thingy.. Doing it in the next push..